sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/trees/TreeNodeSuite.scala

```diff
@@ -885,7 +885,7 @@ class TreeNodeSuite extends SparkFunSuite with SQLHelper {
         |^^^^^
         |""".stripMargin
 
-    assert(origin.context == expected)
+    assert(origin.context.summary == expected)
```

Comment:
Shall we check the other fields of the context like https://github.com/apache/spark/pull/36702/files#diff-83746ef1d50ed7962e780275d0af3e7f9c2e0434b8a75677c269a209f81997adR892 ?

Reply:
I copied the checks from your PR.